lib/socketset/socketset.go

```diff
@@ -0,0 +1,218 @@
+// Mgmt
+// Copyright (C) 2013-2018+ James Shubin and the project contributors
+// Written by James Shubin <james@shubin.ca> and the project contributors
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+// Package socketset is in API for creating a select style netlink
+// socket to recieve events from the kernel.
+package socketset
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"path"
+	"strings"
+	"syscall"
+
+	"golang.org/x/sys/unix"
+
+	errwrap "github.com/pkg/errors"
+)
+
```

Comment:
It's best to re-order the entries in this file. Usually, first we see the `type SocketSet struct`, then it's `New` constructor, and then the methods on that struct in the approximate order they're used in.

And then the dependency struct we use underneath it in the same order.